packages/fuels-programs/src/contract.rs

```diff
@@ -219,4 +115,236 @@ mod tests {
         )
         .unwrap()
     }
+
+    #[test]
+    fn blob_size_must_be_greater_than_zero() {
+        // given
+        let contract = Contract::regular(vec![0x00], Salt::zeroed(), vec![]);
+
+        // when
+        let err = contract
+            .convert_to_loader(0)
+            .expect_err("should have failed because blob size is 0");
+
+        // then
+        assert_eq!(
+            err.to_string(),
+            "blob size must be greater than 0".to_string()
+        );
+    }
+
+    #[test]
+    fn contract_with_no_code_cannot_be_turned_into_a_loader() {
+        // given
+        let contract = Contract::regular(vec![], Salt::zeroed(), vec![]);
+
+        // when
+        let err = contract
+            .convert_to_loader(100)
+            .expect_err("should have failed because there is no code");
+
+        // then
+        assert_eq!(
+            err.to_string(),
+            "must provide at least one blob".to_string()
+        );
+    }
+
+    #[test]
+    fn loader_needs_at_least_one_blob() {
+        // given
+        let no_blobs = vec![];
+
+        // when
+        let err = Contract::loader_for_blobs(no_blobs, Salt::default(), vec![])
+            .expect_err("should have failed because there are no blobs");
+
+        // then
+        assert_eq!(
+            err.to_string(),
+            "must provide at least one blob".to_string()
+        );
+    }
+
+    #[test]
+    fn loader_requires_all_except_the_last_blob_to_be_word_sized() {
+        // given
+        let blobs = [vec![0; 9], vec![0; 8]].map(Blob::new).to_vec();
+
+        // when
+        let err = Contract::loader_for_blobs(blobs, Salt::default(), vec![])
+            .expect_err("should have failed because the first blob is not word-sized");
+
+        // then
+        assert_eq!(
+            err.to_string(),
+            "blob 1/2 has a size of 9 bytes, which is not a multiple of 8".to_string()
+        );
+    }
+
+    #[test]
+    fn last_blob_in_loader_can_be_unaligned() {
+        // given
+        let blobs = [vec![0; 8], vec![0; 9]].map(Blob::new).to_vec();
```

Comment:
would it make sense to error out here. As this can be represented as two blobs with size 8 and then one blob of size 1. Which means that the last blob can have any size smaller than 8.

Reply:
Um. Not sure I follow? 

The only thing we need to enforce is that all blobs, except the last one, need to be word-aligned. 

This test here shows that we tolerate the last blob to be unaligned. 

There is another test that shows that this cannot happen anywhere else.

Reply:
Thanks for the explanation. This makes sense. Disregard my comment because I understood the test wrongly.